### PR TITLE
[CIVIS-6103] fix missing public keys 

### PR DESCRIPTION
**Pull request by thatguyinabeanie**
AWS Code Build was unable to successfully build the container image.
i initially thought it was due to the debian bookworm base image not having public keys for some of the apt-get dependencies we were trying to install.

@fahaddd-git figured out that the actual cause of the issue was that the AWS Code Build environment was based on an older image. this image did not have references to the debian ftp repository where the public keys are accessible from.

This is what effectively caused the issue.

This PR reflect the changes needed to interface with the updated aws cli and docker cli for the updated version of the AWS Code Build Environment

### CODE REVIEW

File: buildspec/push.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+version: 0.2
+phases:
+  pre_build:
+    commands:
+      - echo Logging in to Amazon ECR...
+      - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION)
```
Comment (fahaddd404) [inline buildspec/push.yaml]: `aws ecr get-login-password --region ${AWS_DEFAULT_REGION}| docker login --username AWS --password-stdin ${REPOSITORY_URI}`

File: buildspec/release.yaml [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 version: 0.2
 phases:
-  pre_build:
-    commands:
-      - echo Logging in to Amazon ECR...
-      - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION)
   build:
     commands:
+      - echo Logging in to Amazon ECR...
+      - aws ecr get-login-password --region ${AWS_DEFAULT_REGION} | docker login --username AWS --password-stdin ${REPOSITORY_URI}
```
Comment (mwpeterson) [inline buildspec/release.yaml:7]: This is better in the `pre-build` phase.

File: buildspec/release.yaml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
       - echo Building the Docker image...
       - PATCH_TAG=${TAG_NAME#"v"} # major.minor.patch
       - MINOR_TAG=${PATCH_TAG%.*} # major.minor
       - MAJOR_TAG=${MINOR_TAG%.*} # major
       - docker build -t ${REPOSITORY_URI}:${PATCH_TAG} -t ${REPOSITORY_URI}:${MINOR_TAG} -t ${REPOSITORY_URI}:${MAJOR_TAG} .
-      - docker push ${REPOSITORY_URI}
+      - docker push ${REPOSITORY_URI}:${PATCH_TAG}
+      - docker push ${REPOSITORY_URI}:${MINOR_TAG}
+      - docker push ${REPOSITORY_URI}:${MAJOR_TAG}
```
Comment (mwpeterson) [inline buildspec/release.yaml]: Now that this is on the new codebuild image with docker 20.10 you can do
```suggestion
      - docker image push --all-tags  ${REPOSITORY_URI}
```

**Review by mwpeterson — CHANGES_REQUESTED** [review]
minor cleanup

File: buildspec/release.yaml (hunk 2 above)
Comment (thatguyinabeanie) [reply buildspec/release.yaml:7]: i had it in the pre-build stage before. when it came to trying to push the image, the login didnt seem to take and failed

Comment (thatguyinabeanie) [reply buildspec/release.yaml:7]: so i had everything be part of the same set of commands. and that did the trick

Comment (mwpeterson) [reply buildspec/release.yaml:7]: Strange.

**Review by mwpeterson — APPROVED** [review]
lgtm

**Pull request merged**